src/main/scala/com/yotpo/metorikku/test/ErrorMsgs.scala

```diff
@@ -0,0 +1,122 @@
+package com.yotpo.metorikku.test
+
+object ErrorType extends Enumeration {
+  val InvalidKeysNonExisting, InvalidKeysNonDefined, DuplicatedResults, DuplicatedResultsHeader,
+      MismatchedKeyResultsExpected, MismatchedKeyResultsActual, MismatchedResultsAllCols, InvalidSchemaResults = Value
+}
+
+
+object ResultsType extends Enumeration {
+  val expected = Value("Expected")
+  val actual = Value("Actual")
+  val noRes = Value("")
+}
+
+
+
+case class ErrorMsgData(errorType: ErrorType.Value, tableName: String,
+                        undefinedCols: List[String], allColsKeys: List[String]
```

Comment:
Please use the the concept of default values in the class then you won't have to send every time all of the defaults in the instantiations below.